src/Pages/SearchPhotoCollections/SearchPhotoCollections.js

```diff
@@ -10,21 +11,28 @@ export default class SearchPhotoCollections extends Component {
 		hasError: false,
 		hasMore: true,
 		page: 1,
+		index: -1,
+		total: 0,
 	};
 	fetchNextPage = () => {
+		if (this.state.photos.length >= this.state.total) {
+			this.setState({ hasMore: false });
+			return;
+		}
```

Comment:
I'm wondering if it's not better to have this method in component did update or when you finish fetching data, and next time, in your fetching, you should check if notHasMore, then return, otherwise, you are always setting the state and rerendering the page